src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -399,18 +509,37 @@ object JoinIndexRule
   private def getLRColumnMapping(
       leftBaseAttrs: Seq[String],
       rightBaseAttrs: Seq[String],
+      lp: Seq[NamedExpression],
+      rp: Seq[NamedExpression],
       condition: Expression): Map[String, String] = {
     extractConditions(condition).map {
       case EqualTo(attr1: AttributeReference, attr2: AttributeReference) =>
+        val attrLeftName = if (lp.nonEmpty) {
```

Comment:
Could you add an internal or util function for this? and add some comments - it's for nested columns & reason